pkg/config/config.go

```diff
@@ -0,0 +1,327 @@
+package config
+
+import (
+	"github.com/golang/protobuf/ptypes/any"
+	log "github.com/sirupsen/logrus"
+	"golang.org/x/net/context"
+
+	api "github.com/osrg/gobgp/api"
+	"github.com/osrg/gobgp/internal/pkg/apiutil"
+	"github.com/osrg/gobgp/internal/pkg/config"
+	"github.com/osrg/gobgp/internal/pkg/table"
+	"github.com/osrg/gobgp/pkg/packet/bgp"
+	"github.com/osrg/gobgp/pkg/server"
+)
+
+func ReadConfigFile(configFile, configType string) (*config.BgpConfigSet, error) {
+	return config.ReadConfigfile(configFile, configType)
```

Comment:
no context?

Reply:
Hi, thank you very much for your review.

I did not include a context in this call because it isn't expected to make any API calls or spawn any threads that would want a context. Its job is to read a config file from disk and return the parsed representation of it. I don't think a consumer of this API would expect to pass a context to it.

Under the hood, this call makes a `filepath.Ext(path)` call to try to detect the file time and then it makes `viper` calls to read the config file and get a` BgpConfigSet` instance. Neither accepts a context.